src/librbd/ImageCtx.cc

```diff
@@ -596,7 +596,7 @@ struct C_InvalidateCache : public Context {
   bool ImageCtx::test_features(uint64_t in_features,
                                const RWLock &in_snap_lock) const
   {
-    assert(snap_lock.is_locked());
+    assert(in_snap_lock.is_locked());
```

Comment:
Thanks for the explanation. It looked like a harmless typo to me given that `in_snap_lock` is never used in the function. Plus it looked a bit unnecessary to accept a lock as a parameter and check if its the lock is already acquired. (I had put up a comment in github regarding this but never published it)

I'll revert this change and resubmit.